client/src/main/scripts/database/SearchDatabaseApi.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * MyTake.org website and tooling.
+ * Copyright (C) 2020 MyTake.org, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact us at team@mytake.org
+ */
+import { Foundation } from "../java2ts/Foundation";
+import { Routes } from "../java2ts/Routes";
+import { Search } from "../java2ts/Search";
+import {
+  fetchFactReturningPromise,
+  postRequest,
+  VideoFactHashMap,
+} from "../utils/databaseAPI";
+import { SearchDatabase } from "./SearchDatabase";
+import { reject } from "lodash";
+
+export class SearchDatabaseApi {
+  private _searchDatabase: SearchDatabase | undefined;
+  private static instance: SearchDatabaseApi | undefined;
+  private constructor(private searchTerm: string) {}
+
+  static getInstance(searchTerm?: string): SearchDatabaseApi {
+    if (!SearchDatabaseApi.instance) {
+      if (!searchTerm) {
+        throw "SearchDatabaseApi: Must provide a searchTerm when creating an instance";
+      }
+      SearchDatabaseApi.instance = new SearchDatabaseApi(searchTerm);
+      return SearchDatabaseApi.instance;
+    }
+    // if (
+    //   SearchDatabaseApi.instance.searchTerm &&
+    //   SearchDatabaseApi.instance.searchTerm !== searchTerm
+    // ) {
+    //   throw "SearchDatabaseApi: cannot change search term";
+    // }
```

Comment:
Tests don't pass with this code right now since we have 2 test cases with 2 different search terms. This is the main reason the PR is in draft state. It probably would have made more sense to push this with this code uncommented and tests that fail.